File: cs 315 hw2/codes/duru_ant.dart
void main(){
    
  // pretest or posttest in while loop
  var myVar = 5;
  var pretest = true;
  print("-----------CHECK IF PRETEST OR POSTTEST FOR WHILE LOOP-----------");
   
  // check if dart is posttest in while loop
  while(myVar >= 10){
    pretest = false;
    print("Dart is posttest for while loop");
  }
   
  // check if dart is pretest in while loop
  if(pretest == true){
    print("Dart is pretest for while lopp");
  }
   
  // pretest or posttest in do-while loop
  var myVar2 = 5;
  var pretest2 = true;
  print("-----------CHECK IF PRETEST OR POSTTEST FOR WHILE LOOP-----------");
  do {
    pretest2 = false;
    print("Dart is posttest for do-while loop");
    myVar2++;
  } while (myVar2 >= 10);

  if (pretest2 == true) {
    print ("Dart is pretest for do-while loop");
  }
   
  // break statement for while loop
  var value = 2;
  var loopCount = 1;
  print("-----------BREAK STATEMENT FOR WHILE LOOP-----------");
  while(value <= 10){
    print("loop: $loopCount , value: $value");
    value++;
    loopCount++;
       
    //check if break
    if(value == 7){
      break;
    }
  }
  print("end of while loop");
  print("loop uses break when the value is $value");
   
  // break statement for do-while loop
  var value2 = 2;
  var loopCount2 = 1;
  print("-----------BREAK STATEMENT FOR DO-WHILE LOOP-----------");
  do{
    print("loop: $loopCount2 , value: $value2");
    value2++;
    loopCount2++;
       
    //check if break
    if(value2 == 5){
      break;
    }
  }while(value2 <= 10);
  print("end of do-while loop");
  print("loop uses break when the value is $value2");
  
  // break statement for for loop
  print("-----------BREAK STATEMENT FOR FOR LOOP-----------");
  var lc = 1;
  for(var i = 3; i < 9; i++){
    print("loop: $lc , value: $i");
    lc++;
    if(i == 8){
      print("value is now $i , loop now uses break");
      break;
    }
  }
  print("end of for loop");
   
  // continue statement for while loop
  var value3 = 2;
  var loopCount3 = 1;
  var cValue;
  print("-----------CONTINUE STATEMENT FOR WHILE LOOP-----------");
  while(value3 <= 10){
    value3++;
      //check if continue
      if(value3 == 7){
        cValue = 7;
        loopCount3++;
        print("skipped when value is 7");
        continue;
      }
      print("loop: $loopCount3 , value: $value3");
      loopCount3++;
  }
  print("end of while loop");
  print("loop uses continue when the value is $cValue");
   
  // continue statement for do-while loop
  var value4 = 2;
  var loopCount4 = 1;
  var cValue2;
  print("-----------CONTINUE STATEMENT FOR DO-WHILE LOOP-----------");
  do { 
    value4++; 
        
    //check if continue
    if (value4 == 5) { 
      cValue2 = 5;
      loopCount4++;
      print("skippen when value is:  $cValue2"); 
      continue; 
    } 
        
    print("loop: $loopCount4 , value: $value4"); 
    loopCount4++;
  } while (value4 <= 10); 
  print("end of do-while loop");
  print("loop uses continue when the value is $cValue2");
  
  // continue statement for for loop
   print("-----------CONTINUE STATEMENT FOR FOR LOOP-----------");
   var lc2 = 1;
   for(var i = 3; i < 9; i++){
     if(i == 5){
       print("value is now $i , loop now uses continue");
       continue;
     }
     print("loop: $lc2 , value: $i");
     lc2++;
   }
   print("end of for loop");
}

File: cs 315 hw2/codes/duru_ant.lua
-- while loop test check
print("-----------CHECK PRETEST OR POSTTEST FOR WHILE LOOP-----------")
i = 3
pretest = true

-- posttest check
while i > 10 do
    pretest = false
    print("Lua is posttest for while loop.")
    i = i - 1
end

-- pretest check
if pretest == true then
    print("Lua is pretest for while loop.")
end

-- repeat until loop test check
print("-----------CHECK PRETEST OR POSTTEST FOR REPEAT UNTIL LOOP-----------")
j = 3
pretest2 = true

-- posttest check
repeat
    j = j + 1
    pretest = false
    print("Lua is posttest for repeat until loop")
until j >= 3

-- pretest check
if pretest == true then
    print("Lua is pretest for repeat until loop")
end

-- break statement for while loop
print("-----------BREAK STATEMENT FOR WHILE LOOP-----------")
var = 3
loopCount = 1
print("before the loop var is: ", var)
while var < 10 do
    print("loop: ", loopCount, " var: ", var)
    var = var + 1
    loopCount = loopCount + 1
    if(var == 8) then
        break
    end
end
print("end of while loop")
print("loop used break when value was: ", var)

-- break statement for repeat until loop
print("-----------BREAK STATEMENT FOR REPEAT UNTIL LOOP-----------")
var2 = 2
print("before the loop var is: ", var2)
loopCount2 = 1
repeat
    print("loop: ", loopCount2, " var: ", var2)
    var2 = var2 + 1
    loopCount2 = loopCount2 + 1
    if var2 == 5 then
        break
    end
until var2 > 10
print("end of repeat until loop")
print("loop used break when value was: ", var2)

-- break statement for for loop
print("-----------BREAK STATEMENT FOR FOR LOOP-----------")
loopCount3 = 1
for i = 3, 10, 1 do
    if i == 6 then
        var3 = i
        break
    end
    print("loop: ", loopCount3, " var: ", i)
    loopCount3 = loopCount3 + 1
end
print("end of for loop")
print("loop used break when value was: ", var3)